mesonbuild/backend/backends.py

```diff
@@ -866,7 +865,10 @@ def object_filename_from_source(self, target: build.BuildTarget, source: 'FileOr
                 gen_source = os.path.relpath(os.path.join(build_dir, rel_src),
                                              os.path.join(self.environment.get_source_dir(), target.get_subdir()))
         machine = self.environment.machines[target.for_machine]
-        return self.canonicalize_filename(gen_source) + '.' + machine.get_object_suffix()
+        if targetdir is None:
+            return self.canonicalize_filename(gen_source) + '.' + machine.get_object_suffix()
+        else:
+            return os.path.join(targetdir, self.canonicalize_filename(gen_source) + '.' + machine.get_object_suffix())
```

Comment:
small cleanliness thing here, could we do something like:
```python
ret = <long concat>
if targetdir is not None:
    return os.path.join(targetdir, ret)
return ret
```

Reply:
@tangalbert919 based on the latest push I think most review comments were addressed, but not this one. Was it overlooked? Do you prefer the existing approach?

Reply:
I overlooked a few of them by mistake and quickly made corrections.

Reply:
Awesome, thanks. LGTM.